src/Controls/src/Core/SafeAreaElement.cs

```diff
@@ -19,6 +22,22 @@ static void OnSafeAreaEdgesChanged(BindableObject bindable, object oldValue, obj
 			if (bindable is IView view)
 			{
 				view.InvalidateMeasure();
+
+#if ANDROID
+				// On Android, request layout does not call OnApplyWindowInsets. so we manually call it.
+				if (bindable is Element element && element.Handler is IElementHandler handler)
```

Comment:
I think you could achieve this with a mapper couldn't you? 

Add a mapper to MapSafeAreaEdges and have a specific one for android that calls this code?

The InvalidateMeasure should probably also be in a mapper